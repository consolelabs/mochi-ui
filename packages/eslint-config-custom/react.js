const { resolve } = require('node:path')

const project = resolve(process.cwd(), 'tsconfig.json')

const dfLintConfig = require.resolve('@dwarvesf/react-eslint-config')

module.exports = {
  extends: [dfLintConfig, 'eslint-config-turbo'],
  plugins: ['prettier'],
  parserOptions: {
    project,
  },
  globals: {
    JSX: true,
  },
  ignorePatterns: [
    'node_modules/',
    'dist/',
    '.eslintrc.js',
    '**/*.css',
    '*.snap',
  ],
  settings: {
    'import/resolver': {
      typescript: {
        project,
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
  rules: {
    'react/function-component-definition': 'off',
    'turbo/no-undeclared-env-vars': 'off',
    'react/jsx-key': 'off',
    'no-dupe-class-members': 'off',
    'prefer-arrow-callback': 'off',
    'import/export': 'off',
    'prettier/prettier': [
      'error',
      {
        bracketSpacing: true,
        jsxBracketSameLine: false,
        jsxSingleQuote: false,
        printWidth: 80,
        proseWrap: 'always',
        semi: false,
        singleQuote: true,
        tabWidth: 2,
        trailingComma: 'all',
        xmlWhitespaceSensitivity: 'ignore',
      },
    ],
  },
  overrides: [
    {
      parser: '@typescript-eslint/parser',
      files: ['**/*.ts', '**/*.tsx'],
      plugins: ['@typescript-eslint'],
      rules: {
        'no-use-before-define': 'off',
        'react/react-in-jsx-scope': 'off',
        'react/jsx-sort-props': 'off',
        'no-void': 'off',
        'no-unused-vars': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_',
            caughtErrorsIgnorePattern: '^_',
          },
        ],
        '@typescript-eslint/no-use-before-define': [
          'error',
          { functions: false, classes: false },
        ],
      },
    },
  ],
}
