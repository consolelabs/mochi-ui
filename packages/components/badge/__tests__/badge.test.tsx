import { render } from '@testing-library/react'
import Badge from '../src/badge'

describe('Badge', () => {
  it('renders correctly with label only', () => {
    const { container } = render(<Badge label="Label" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with icon only', () => {
    const { container } = render(<Badge icon={<span>Icon</span>} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with label and icon', () => {
    const { container } = render(
      <Badge icon={<span>Icon</span>} label="Label" />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with icon on the right', () => {
    const { container } = render(
      <Badge icon={<span>Icon</span>} label="Label" iconPosition="right" />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with custom class name', () => {
    const { container } = render(
      <Badge icon={<span>Icon</span>} label="Label" className="custom-class" />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with custom icon class name', () => {
    const { container } = render(
      <Badge
        icon={<span>Icon</span>}
        label="Label"
        iconClassName="custom-icon-class"
      />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with avatar icon', () => {
    const { container } = render(
      <Badge icon={<span>Icon</span>} label="Label" isAvatarIcon />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with appearance', () => {
    const { container } = render(
      <Badge icon={<span>Icon</span>} label="Label" appearance="primary" />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
