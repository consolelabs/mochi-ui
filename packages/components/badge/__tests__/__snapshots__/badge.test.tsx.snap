// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly with appearance 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pl-[6px]"
>
  <span
    class="flex text-primary-500"
  >
    <span>
      Icon
    </span>
  </span>
  Label
</span>
`;

exports[`Badge renders correctly with avatar icon 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pl-[6px] !pl-[3px] gap-[6px]"
>
  <span
    class="flex text-primary-500 w-4 h-4"
  >
    <span>
      Icon
    </span>
  </span>
  Label
</span>
`;

exports[`Badge renders correctly with custom class name 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pl-[6px] custom-class"
>
  <span
    class="flex text-primary-500"
  >
    <span>
      Icon
    </span>
  </span>
  Label
</span>
`;

exports[`Badge renders correctly with custom icon class name 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pl-[6px]"
>
  <span
    class="flex text-primary-500 custom-icon-class"
  >
    <span>
      Icon
    </span>
  </span>
  Label
</span>
`;

exports[`Badge renders correctly with icon on the right 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pr-[6px]"
>
  Label
  <span
    class="flex text-primary-500"
  >
    <span>
      Icon
    </span>
  </span>
</span>
`;

exports[`Badge renders correctly with icon only 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pl-[6px] !p-1"
>
  <span
    class="flex text-primary-500 text-primary-700"
  >
    <span>
      Icon
    </span>
  </span>
</span>
`;

exports[`Badge renders correctly with label and icon 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pl-[6px]"
>
  <span
    class="flex text-primary-500"
  >
    <span>
      Icon
    </span>
  </span>
  Label
</span>
`;

exports[`Badge renders correctly with label only 1`] = `
<span
  class="rounded-2xl py-0.5 text-xs font-medium flex items-center justify-center leading-[18px] px-2 gap-1 bg-primary-100 text-primary-700 pl-[6px] !px-2"
>
  Label
</span>
`;
